pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -638,16 +910,138 @@ func (pl *dynamicResources) PreFilter(ctx context.Context, state *framework.Cycl
+	if class.ParametersRef == nil {
+		return nil, nil
+	}
+
+	if class.ParametersRef.APIGroup == resourcev1alpha2.SchemeGroupVersion.Group &&
+		class.ParametersRef.Kind == "ResourceClassParameters" {
+		// Use the parameters which were referenced directly.
+		parameters, err := pl.classParametersLister.ResourceClassParameters(class.ParametersRef.Namespace).Get(class.ParametersRef.Name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return nil, statusUnschedulable(logger, fmt.Sprintf("class parameters %s not found", klog.KRef(class.ParametersRef.Namespace, class.ParametersRef.Name)))
+			}
+			return nil, statusError(logger, fmt.Errorf("get class parameters %s: %v", klog.KRef(class.Namespace, class.ParametersRef.Name), err))
+		}
+		return parameters, nil
+	}
+
+	allParameters, err := pl.classParametersLister.ResourceClassParameters(class.Namespace).List(labels.Everything())
+	if err != nil {
+		return nil, statusError(logger, fmt.Errorf("listing class parameters failed: %v", err))
+	}
+	for _, parameters := range allParameters {
+		if parameters.GeneratedFrom == nil {
+			continue
+		}
+		if parameters.GeneratedFrom.APIGroup == class.ParametersRef.APIGroup &&
+			parameters.GeneratedFrom.Kind == class.ParametersRef.Kind &&
+			parameters.GeneratedFrom.Name == class.ParametersRef.Name &&
+			parameters.GeneratedFrom.Namespace == class.ParametersRef.Namespace {
+			return parameters, nil
+		}
+	}
+	return nil, statusUnschedulable(logger, fmt.Sprintf("generated class parameters for %s.%s %s not found", class.ParametersRef.Kind, class.ParametersRef.APIGroup, klog.KRef(class.Namespace, class.ParametersRef.Name)))
+}
+
+func (pl *dynamicResources) lookupClaimParameters(logger klog.Logger, claim *resourcev1alpha2.ResourceClaim) (*resourcev1alpha2.ResourceClaimParameters, *framework.Status) {
+	if claim.Spec.ParametersRef == nil {
+		return nil, nil
+	}
+	if claim.Spec.ParametersRef.APIGroup == resourcev1alpha2.SchemeGroupVersion.Group &&
+		claim.Spec.ParametersRef.Kind == "ResourceClaimParameters" {
+		// Use the parameters which were referenced directly.
+		parameters, err := pl.claimParametersLister.ResourceClaimParameters(claim.Namespace).Get(claim.Spec.ParametersRef.Name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return nil, statusUnschedulable(logger, fmt.Sprintf("claim parameters %s not found", klog.KRef(claim.Namespace, claim.Spec.ParametersRef.Name)))
+			}
+			return nil, statusError(logger, fmt.Errorf("get claim parameters %s: %v", klog.KRef(claim.Namespace, claim.Spec.ParametersRef.Name), err))
+		}
+		return parameters, nil
+	}
+
+	allParameters, err := pl.claimParametersLister.ResourceClaimParameters(claim.Namespace).List(labels.Everything())
+	if err != nil {
+		return nil, statusError(logger, fmt.Errorf("listing claim parameters failed: %v", err))
+	}
+	for _, parameters := range allParameters {
```

Comment:
This seems equally unintuitive as above, although it's harder for end users to know how to reference a ResourceClaimParameters, instead of the CRD.

Just to confirm, @johnbelamaric, does this match your expectations?

Reply:
I think we will work through some of this and likely improve the UX in subsequent iterations. Ideally, users will write resource claims in a K8s native way, and only layer in CRs for config parameters. But Patrick has pointed out that this may be harder in some cases. So, it's still something of an open question but should not block the PR.

Reply:
Having the ability to provide a vendor specific CRD is not just a "nice-to-have" but actually a requirement in terms of being able to do pass through per-device configuration. There have been arguments against it for "selection" of devices (which I disagree with), but I see no way around it for configuration of devices -- only the vendor can know what the APIs for this configuration will be and trying to shoehorn it into something that can be generalized doesn't seem practical.

Reply:
Acceptable, as per the comment in the other thread.

Reply:
@klueska yep, that's what I meant by "only layer in CRs for config parameters". But we will iterate and come to some common understanding hopefully in 1.31 :)

Reply:
I think we posted our comments at the same time, so I hadn't seen yours yet. We basically said the same thing :)